DragaliaAPI/Features/Missions/FortMissionProgressionService.cs

```diff
@@ -0,0 +1,86 @@
+﻿using DragaliaAPI.Features.Fort;
+using DragaliaAPI.Shared.Definitions.Enums;
+using DragaliaAPI.Shared.MasterAsset.Models.Missions;
+using Microsoft.EntityFrameworkCore;
+
+namespace DragaliaAPI.Features.Missions;
+
+public class FortMissionProgressionService(
+    IFortRepository fortRepository,
+    IMissionProgressionService missionProgressionService
+) : IFortMissionProgressionService
+{
+    public async Task<int> GetTotalFortLevel()
+    {
+        return await fortRepository.Builds
+                .Where(x => x.PlantId != FortPlants.TheHalidom)
+                .SumAsync(
+                    x => x.BuildEndDate == DateTimeOffset.UnixEpoch ? x.Level : (int?)x.Level - 1
```

Comment:
Why are you casting level to nullable and coalescing this? Does Sum throw if the sequence is empty?

Reply:
correct, and if the type is int? it returns null instead